From abaabedbd058840262d1f7eb23a12167f37aa79a Mon Sep 17 00:00:00 2001
Message-Id: <abaabedbd058840262d1f7eb23a12167f37aa79a.1430046936.git.chang-joon.lee@intel.com>
In-Reply-To: <6f7782ceadb10df792cdc4378455d6e03124c770.1430046936.git.chang-joon.lee@intel.com>
References: <6f7782ceadb10df792cdc4378455d6e03124c770.1430046936.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 3 Mar 2015 15:16:59 +0530
Subject: [PATCH 4/4] FOR_UPSTREAM [VPG]: drm/i915/chv: Enable
 WaDisableShadowRegForCpd

This WA disable usage of shadow register during CPD/RC6 transactions on CHV

Issue: GMINL-6773
Change-Id: Ie6be000aac4586caba9389d8e1fdbc8354e873cd
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    1 +
 drivers/gpu/drm/i915/intel_pm.c |    4 ++++
 2 files changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 79beb83..2104ee6 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -6017,6 +6017,7 @@ enum punit_power_well {
 #define  GTFIFOCTL				0x120008
 #define    GT_FIFO_FREE_ENTRIES_MASK		0x7f
 #define    GT_FIFO_NUM_RESERVED_ENTRIES		20
+#define    GT_FIFO_CTL_BLOCK_POLICY		(3<<11)
 
 #define  HSW_IDICR				0x9008
 #define    IDIHASHMSK(x)			(((x) & 0x3f) << 16)
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index e18161d..af252f5 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -6741,6 +6741,10 @@ static void cherryview_init_clock_gating(struct drm_device *dev)
 	/* WaDisableSDEUnitClockGating:chv */
 	I915_WRITE(GEN8_UCGCTL6, I915_READ(GEN8_UCGCTL6) |
 		   GEN8_SDEUNIT_CLOCK_GATE_DISABLE);
+
+	/* WaDisableShadowRegForCpd */
+	I915_WRITE(GTFIFOCTL, I915_READ(GTFIFOCTL) |
+			      GT_FIFO_CTL_BLOCK_POLICY);
 }
 
 static void g4x_init_clock_gating(struct drm_device *dev)
-- 
1.7.9.5

